e2etest/bgp_test.go

```diff
@@ -945,6 +947,74 @@ var _ = ginkgo.Describe("BGP", func() {
 						},
 					},
 				}}, "ipv6"))
+
+		table.DescribeTable("configure bgp local-preference and verify it gets propagated",
+			func(addressPools []config.AddressPool, ipFamily string) {
+				configData := config.File{
+					Peers: peersForContainers(frrContainers, ipFamily),
+					Pools: addressPools,
+				}
+				for _, c := range frrContainers {
+					pairExternalFRRWithNodes(cs, c, ipFamily)
+				}
+
+				err := configUpdater.Update(configData)
+				framework.ExpectNoError(err)
+
+				for _, c := range frrContainers {
+					validateFRRPeeredWithNodes(cs, c, ipFamily)
+				}
+
+				svc, _ := createServiceWithBackend(cs, f.Namespace.Name, "external-local-lb", testservice.TrafficPolicyCluster) // Is a sleep required here?
+
+				defer func() {
+					err := cs.CoreV1().Services(svc.Namespace).Delete(context.TODO(), svc.Name, metav1.DeleteOptions{})
+					framework.ExpectNoError(err)
+				}()
+
+				for _, i := range svc.Status.LoadBalancer.Ingress {
+					ginkgo.By("validate LoadBalancer IP is in the AddressPool range")
+					ingressIP := e2eservice.GetIngressPoint(&i)
+					err = validateIPInRange(addressPools, ingressIP)
+					framework.ExpectNoError(err)
+				}
+
+				allNodes, err := cs.CoreV1().Nodes().List(context.Background(), metav1.ListOptions{})
+				framework.ExpectNoError(err)
+
+				for _, c := range frrContainers {
+					validateService(cs, svc, allNodes.Items, c)
+					Eventually(func() error {
+						return frr.RoutesMatchedLocalPref(c, IPv4LocalPref, IPv6LocalPref)
```

Comment:
I'd rather have the localpref we are matching against passed as a parameter.
Here we match the global constant against the other global constant being set in the table entry, which are incidentally the same. I think it's better to have it more explicit to be the same value..